### PR TITLE
Restructure `pairing.py` function

### PR DESCRIPTION
**Pull request by rmatsum836**
## Description
In order to accommodate the `_check_validity` function `generate_indirect_connectivity` is moved into a private function.  `new_generate_indirect` is added to `pairing.py` to calculate the indirect connectivity matrix until the previous matrix equals the current matrix.  

NOT READY TO MERGE

### CODE REVIEW

**Comment by rmatsum836** [comment]
Okay after undoing most of my screwups the overall changes are as follows:
- `generate_indirect_connectivity` is turned into an internal function `_generate_indirect_connectivity`
- The function `new_generate_indirect` calls `_generate_indirect_connectivity` and will keep calculating the indirect connectivity matrix until `_check_validity` becomes True

**Comment by rmatsum836** [comment]
Going to write tests for this and then should be okay to merge.

File: pairing/pairing.py [hunk 1]
```diff
@@ -155,9 +154,18 @@ def _check_validity(c_I):
...
     """
 
-    test_indirect = generate_indirect_connectivity(c_I)
+    test_indirect = _generate_indirect_connectivity(c_I)
     return (test_indirect == c_I).all()
 
+def new_generate_indirect(direct_corr):
+    new_indirect = _generate_indirect_connectivity(
+            direct_corr)
+    while _check_validity(new_indirect) == False:
```
Comment (mattwthompson) [inline pairing/pairing.py]: Nice! Have you seen any performance dip from this? I don't know if we have cases requiring multiple calls

Comment (rmatsum836) [reply pairing/pairing.py]: I'll run some tests to see

**Comment by rmatsum836** [comment]
Just added a description for the new function, and added two lines between each function per PEP8.  Anything else before merging?

**Comment by rmatsum836** [comment]
One of the appveyor tests failed, I believe for the same reason we couldn't figure out a few days ago.

**Comment by mattwthompson** [comment]
Thanks!

**Pull request merged**